x/mongo/driver/xoptions/options.go

```diff
@@ -31,6 +32,12 @@ func SetInternalClientOptions(opts *options.ClientOptions, key string, option an
 			return fmt.Errorf(typeErr, key)
 		}
 		opts.Deployment = d
+	case "authenticateToAnything":
+		b, ok := option.(bool)
+		if !ok {
+			return fmt.Errorf(typeErr, key)
```

Comment:
Can we update these errors to describe the type that we want? Possible using some kind of composing function:

```go
typeErrFuc := func() string {
	return fmt.Sprintf("unexecpted type for %q, wanted %T, got %T", key, option, option)
}
```